cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/PredefinedAutoScaling.java

```diff
@@ -157,7 +159,9 @@ AutoscalingPolicyServiceClient getAutoscalingPolicyServiceClient()
     throws IOException {
     CredentialsProvider credentialsProvider = FixedCredentialsProvider.create(dataprocConf.getDataprocCredentials());
 
-    String regionalEndpoint = dataprocConf.getRegion() + "-" + dataprocConf.getRootUrl();
+    String rootUrl = Optional.ofNullable(dataprocConf.getRootUrl())
```

Comment:
Can this logic move inside the DataprocConf class? Seems like it is common in multiple places.